src/helpers/error.ts

```diff
@@ -0,0 +1,31 @@
+import { NonRetriableError } from "../components/NonRetriableError";
+
+export const marshalError = (err: unknown): Record<string, any> | string => {
+  if (err instanceof NonRetriableError) {
+    return {
+      message: err.message,
+      stack: err.stack,
+      name: err.name,
+      cause: err.cause
+        ? err.cause instanceof Error
+        ? marshalError(err.cause)
+        : JSON.stringify(err.cause)
+          : undefined,
+    };
+  }
+
+  if (err instanceof Error) {
+    return {
+      message: err.message,
+      stack: err.stack,
+      name: err.name,
+    }
+  }
+
+  return `Unknown error: ${JSON.stringify(err)}`
```

Comment:
What about returning this string as the message parameter so it's more consistent in return type?

```suggestion
  return {
    message: `Unknown error: ${JSON.stringify(err)}`
  }
```